### PR TITLE
Added TSLint

### PR DESCRIPTION
**Pull request by t-sauer**
I started with the recommended preset and just changed rules to match the current style or which would lead to way too complex changes (e.g. `max-classes-per-file`).  
If you want any other rules I would be glad to change them.

### CODE REVIEW

File: src/application.ts [hunk 1]
```diff
@@ -64,17 +64,17 @@ export default class Application implements Owner {
...
   /** @hidden */
-  initRegistry(): void {
-    let registry = this._registry = new Registry();
+  public initRegistry(): void {
+    const registry = this._registry = new Registry();
 
     // Create ApplicationRegistry as a proxy to the underlying registry
     // that will only be available during `initialize`.
-    let appRegistry = new ApplicationRegistry(this._registry, this.resolver);
+    const appRegistry = new ApplicationRegistry(this._registry, this.resolver);
```
Comment (locks) [inline src/application.ts]: Hm, do we also follow "liberally let" in TS?

Comment (rwjblue) [inline src/application.ts]: yes, please

**Comment by t-sauer** [comment]
I disabled the `prefer-const` rule.

**Comment by dgeb** [comment]
@t-sauer Thanks for your work here.

We have a few things to discuss and decide upon as a team:

* [member-access](https://palantir.github.io/tslint/rules/member-access/) - I recommend "no-public", which forbids public accessibility to be specified, because this is the default. (This is the opposite approach to that taken in this PR, which I find a bit noisy).

* Our browser support matrix for Glimmer.js, since that will drive usage of symbols, which will drive support for language features like `for ... of`.

We'll try to get back to you soon to minimize churn.

**Comment by t-sauer** [comment]
Seems like TSLint was already added so this PR isn't needed anymore.

**Pull request closed without merging**